### PR TITLE
Code update: BinaryTree Topologies`

### PR DESCRIPTION
**Pull request by maxsecure12**
updated code for BinaryTree Topology

### CODE REVIEW

File: Int.ipynb [hunk 1]
```diff
@@ -8578,6 +8590,40 @@
...
+        "Number of Binary Tree Topologies : L 155"
+      ],
+      "metadata": {
+        "id": "YBG4HAgP_YGC"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def numberOfBinaryTreeTopologies(n): #   O((n(2n)!/n!(n+1)!))   | O(n) \n",
```
Comment (FaisalWant) [inline Int.ipynb:8606]: update with the cached implemenation

**Review by FaisalWant — COMMENTED** [review]
update the code for the cached Implementation.

File: Int.ipynb (hunk 1 above)
Comment (maxsecure12) [reply Int.ipynb:8606]: sure thanks for pointing this out 👍 will update it right now

File: Int.ipynb [hunk 2]
```diff
@@ -8617,7 +8617,24 @@
         "    numberofTrees+= numberOfLeftTrees * numberOfRightTrees \n",
         "\n",
         "  return numberOfTrees \n",
-        "\n"
+        "\n",
+        "\n",
+        "\n",
+        "#----------------------------------------------------\n",
```
Comment (FaisalWant) [inline Int.ipynb:8623]: try iterative solution

**Review by FaisalWant — COMMENTED** [review]
try to implement iterative solution for it.

**Review by FaisalWant — APPROVED** [review]
This looks good....

**Pull request merged**